### PR TITLE
[MAINTENANCE] Additional examples of `row_condition` for Spark and SQL

### LINKED ISSUE #7531: row_condition implementation issues with pyspark
**Describe the bug**
"row_condition" has implementation issues with pyspark.

**To Reproduce**
1. Create an expectation with a row_condition (with pyspark). For example: "row_condition": ‘col(“id”) != “1234"’
2. In the results json (or the format you select), the error "unable to parse condition" will raise in a "expection_info"

**Expected behavior**
The result should be conditioned.

**Environment (please complete the following information):**
 - Operating System: Windows
 - Great Expectations Version: [e.g. 0.13.2]

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
# What does this PR do? 
- Closes #7531
- Adds example to Documentation for valid `row_conditions` values for Spark and SQL, which was the cause of the issue (particularly BigQuery). 
- Fix for `_sqlalchemy_column_map_condition_values()` for SqlAlchemy > 2.0
- Tests in `tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py`, which test for 2 cases of `row_condition` values, and the 3 ways that we can specify a Batch for Sql (`table_name`, Selectable and query). 

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

### CODE REVIEW

File: tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    )
+    desired_metric.metric_dependencies = {"table.columns": table_columns_metric}
+    results = execution_engine.resolve_metrics(metrics_to_resolve=(desired_metric,))
+    metrics = dict()
+    metrics = {
+        "unexpected_condition": results[desired_metric.id],
+        "table.columns": table_column_metrics_results[table_columns_metric.id],
+    }
+    mp = MapMetricProvider()
+    res = _sqlalchemy_column_map_condition_values(
```
Comment (Shinnnyshinshin) [inline tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py]: Method that was changed for this PR

File: docs/docusaurus/docs/reference/expectations/conditional_expectations.md [hunk 2]
```diff
@@ -115,6 +115,25 @@ PClass=="1st"
 """  # never use \n inside !!!
 ```
 
+For Spark and SQL, you will also want to specify your columns using the `col()` function. 
+
+Some examples of valid `row_condition` values include: 
+
+```python
+row_condition="col(\"foo\") == \"Two  Two\""  # foo is 'Two Two'
```
Comment (rreinoldsc) [inline docs/docusaurus/docs/reference/expectations/conditional_expectations.md]: nit: Is this valid?
```suggestion
row_condition='col("foo") == "Two  Two"'  # foo is 'Two Two'
```

Comment (rreinoldsc) [reply docs/docusaurus/docs/reference/expectations/conditional_expectations.md]:
```
"col(\"foo\") == \"Two Two\""=='col("foo") == "Two Two"'
```

File: great_expectations/expectations/metrics/map_metric_provider/column_map_condition_auxilliary_methods.py [hunk 3]
```diff
@@ -271,8 +272,12 @@ def _sqlalchemy_column_map_condition_values(
         unexpected_condition
     )
     if not _is_sqlalchemy_metric_selectable(map_metric_provider=cls):
-        query = query.select_from(selectable)
-
+        if version.parse(sa.__version__) >= version.parse("1.4") and hasattr(
```
Comment (anthonyburdi) [inline great_expectations/expectations/metrics/map_metric_provider/column_map_condition_auxilliary_methods.py]:
```suggestion
        if version.parse(sa.__version__) >= version.parse("2.0") and hasattr(
```

File: tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    batch_data = SqlAlchemyBatchData(
+        execution_engine=execution_engine,
+        selectable=sa.select(sa.text("*")).select_from(sa.table("test_table")),
+    )
+    execution_engine.load_batch_data("__", batch_data)
+    return execution_engine
+
+
+@pytest.mark.big
+@pytest.mark.parametrize(
```
Comment (anthonyburdi) [inline tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py:144]: Suggestion - two parametrizes. One for execution engine fixture name, the other for the rest.

File: tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+                "condition_parser": "great_expectations__experimental__",
+            },
+            [14.8],
+        ),
+    ],
+)
+def test_column_map_condition_values_row_condition(
+    execution_engine_fixture_name, metric_domain_kwargs, expected_result, request, sa
+):
+    execution_engine = request.getfixturevalue(execution_engine_fixture_name)
```
Comment (anthonyburdi) [inline tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py:251]: :heart:

File: tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+            {
+                "column": "total_amount",
+                "row_condition": 'col("pk_1")!=0',
+                "condition_parser": "great_expectations__experimental__",
+            },
+            [14.8],
+        ),
+    ],
+)
+def test_column_map_condition_values_row_condition(
```
Comment (anthonyburdi) [inline tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py]: Consider adding spark tests to `tests/execution_engine/test_sparkdf_execution_engine.py`

File: tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+        "parse_strings_as_datetimes": False,
+        "allow_cross_type_comparisons": None,
+        "result_format": {
+            "result_format": "COMPLETE",
+            "partial_unexpected_count": 20,
+            "include_unexpected_rows": False,
+        },
+    }
+
+    table_columns_metric, table_column_metrics_results = get_table_columns_metric(
```
Comment (anthonyburdi) [inline tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py]: From discussion: Move after 194

Comment (anthonyburdi) [reply tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py]: Please add a short comment describing why we need to pre compute the metric dependency.

**Review by anthonyburdi — COMMENTED** [review]
Thanks for the synchronous review!

File: great_expectations/expectations/metrics/map_metric_provider/column_map_condition_auxilliary_methods.py (hunk 3 above)
Comment (Shinnnyshinshin) [reply great_expectations/expectations/metrics/map_metric_provider/column_map_condition_auxilliary_methods.py]: Turns out this is `1.4`. With `SQLAlchemy==1.4.49` we get the following error : 

```bash
FAILED tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py::test_column_map_condition_values_row_condition[test_backends0-metric_domain_kwargs0-expected_result0-execution_engine_with_mini_taxi_table_name] - sqlalchemy.exc.SADeprecationWarning: Implicit coercion of SELECT and textual SELECT constructs into FROM clauses is deprecated; please call .subquery() on any Core select or ORM Query object in order to produce a subquery object.
```

File: docs/docusaurus/docs/reference/expectations/conditional_expectations.md (hunk 2 above)
Comment (Shinnnyshinshin) [reply docs/docusaurus/docs/reference/expectations/conditional_expectations.md]: done! thank you for this.

File: tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py (hunk 7 above)
Comment (Shinnnyshinshin) [reply tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py]: Done

File: tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py (hunk 4 above)
Comment (Shinnnyshinshin) [reply tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py:144]: Done! And thank you for the suggestion here. It simplified the parameterization greatly

File: tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py (hunk 6 above)
Comment (Shinnnyshinshin) [reply tests/expectations/metrics/column_map_metrics/test_column_map_condition_auxillary_methods.py]: Done. Added equivalent spark test :)

File: great_expectations/expectations/metrics/map_metric_provider/column_map_condition_auxilliary_methods.py (hunk 3 above)
Comment (anthonyburdi) [reply great_expectations/expectations/metrics/map_metric_provider/column_map_condition_auxilliary_methods.py]: Oh interesting. We don't support sqlalchemy < 1.4 anymore (https://github.com/great-expectations/great_expectations/blob/develop/reqs/requirements-dev-lite.txt#L21). What do you think about just removing the if/else and only using the >= 1.4 version?

**Review by anthonyburdi — COMMENTED** [review]
Looks good! Just one comment to discuss. Thanks for making these changes!

**Review by anthonyburdi — APPROVED** [review]
LGTM! Thanks for working through this.

**Pull request merged**